fh-http/src/manager/request_conversation.rs

```diff
@@ -0,0 +1,133 @@
+use super::{request_processor::get_request_processor, RequestProcessorError};
+use anyhow::Result;
+use chrono::{TimeZone, Utc};
+use sqlx::{pool::PoolConnection, Sqlite};
+use std::str::FromStr;
+use uuid::Uuid;
+
+pub(crate) enum AuditEntry {
+    Request(fh_v8::request::Request),
+    Response(fh_v8::response::Response),
+    Log(String),
+}
+
+pub(crate) struct RequestConversation {
+    id: Uuid,
+    created_at: chrono::DateTime<Utc>,
+    request_processor_id: Uuid,
+    audit_entries: Vec<AuditEntry>,
+}
```

Comment:
This is a nice data model. Let me just share some thoughts.

As far as the current design of the whole machinery goes, we will have a single main request/response conversation. That is, the ingress HTTP request into the public endpoint and the final response made with `respond_with` (#20).

On the other hand, during the execution of the recipe, different request/response conversations can be made on behalf of `dispatch_request` (#25), right?

So, we might want to discriminate between both variants here. I am proposing a `main` slot for the embracing HTTP conversation and an `items` slot for the inner HTTP conversations.

In order to give immediate feedback to the user, we want to fill in the slots early throughout the execution of the recipe. That is:

1. Request comes in.
2. Create `RequestConversation` instance.
3. Fill in `main.request`.
4. On `dispatch_request`, fill in `items[0].request`.
5. When receiving the response, fill in `items[0].response`.
6. Optionally repeat 4. and 5. for more invocations of `dispatch_request`.
7. Before or after finishing the recipe through an explicit or implict `respond_with`, fill in `main.response`.

What do you think about this?

```suggestion
pub(crate) struct RequestConversation {
    id: Uuid,
    created_at: chrono::DateTime<Utc>,
    request_processor_id: Uuid,
    main: AuditEntry,
    items: Vec<AuditEntry>,
}
```

P.S.: Shouldn't `AuditEntry` also be a `struct` to have the convenience of named attributes in there?

Reply:
We can have named attributes in the enum variants, too - rust offers that and defines it as idiomatic. So i can change that, but its fone to do that later when we need it

Reply:
For now, I kept the design a bit simpler. We only store a Vec<AuditItem> in the `RequestConversation` struct. For now we have a convention, that the first entry is always the incoming request. I implemented a `parent` pattern, which we can use to link responses to the incoming request or self-issued requests. This way we couple Requests and Responses together. That's all I can do so far, here.